### PR TITLE
UserCard: Small Gtk4 prep

### PR DESCRIPTION
**Pull request by lenemter**

### CODE REVIEW

File: src/Cards/UserCard.vala [hunk 1]
```diff
@@ -123,36 +116,34 @@ public class Greeter.UserCard : Greeter.BaseCard {
...
-        form_grid.add (login_stack);
-        form_grid.add (session_button);
+        form_box.add (login_stack);
+        form_box.add (session_button);
 
         form_revealer = new Gtk.Revealer () {
             margin_bottom = 12,
             reveal_child = true,
-            transition_type = Gtk.RevealerTransitionType.SLIDE_DOWN
+            transition_type = Gtk.RevealerTransitionType.SLIDE_DOWN,
```
Comment (Marukesu) [inline src/Cards/UserCard.vala]:
```suggestion
            transition_type = SLIDE_DOWN,
```

File: src/Cards/UserCard.vala [hunk 2]
```diff
@@ -123,36 +116,34 @@ public class Greeter.UserCard : Greeter.BaseCard {
...
         login_stack.add_named (password_grid, "password");
         login_stack.add_named (login_button, "button");
-        login_stack.add_named (disabled_grid, "disabled");
+        login_stack.add_named (disabled_box, "disabled");
 
-        var form_grid = new Gtk.Grid () {
-            column_spacing = 6,
-            margin = 24,
-            margin_bottom = 12,
+        var form_box = new Gtk.Box (Gtk.Orientation.HORIZONTAL, 6) {
```
Comment (Marukesu) [inline src/Cards/UserCard.vala]:
```suggestion
        var form_box = new Gtk.Box (HORIZONTAL, 6) {
```

File: src/Cards/UserCard.vala [hunk 3]
```diff
@@ -123,36 +116,34 @@ public class Greeter.UserCard : Greeter.BaseCard {
 
         var disabled_message = new Gtk.Label (_("Account disabled"));
 
-        var disabled_grid = new Gtk.Grid () {
-            column_spacing = 6,
+        var disabled_box = new Gtk.Box (Gtk.Orientation.HORIZONTAL, 6) {
             halign = Gtk.Align.CENTER,
```
Comment (Marukesu) [inline src/Cards/UserCard.vala]:
```suggestion
        var disabled_box = new Gtk.Box (HORIZONTAL, 6) {
            halign = CENTER,
```

File: src/Cards/UserCard.vala [hunk 4]
```diff
@@ -204,16 +194,19 @@ public class Greeter.UserCard : Greeter.BaseCard {
...
+            margin_bottom = 6,
+            margin_start = 6,
+            margin_end = 6,
+            loadable_icon = new FileIcon (File.new_for_path (lightdm_user.image))
         };
-        avatar.loadable_icon = new FileIcon (File.new_for_path (lightdm_user.image));
 
         var avatar_overlay = new Gtk.Overlay () {
             halign = Gtk.Align.CENTER,
+            valign = Gtk.Align.START,
```
Comment (Marukesu) [inline src/Cards/UserCard.vala]:
```suggestion
            halign = CENTER,
            valign = START,
```

File: src/Cards/UserCard.vala [hunk 5]
```diff
@@ -1,20 +1,6 @@
...
- * This program is distributed in the hope that it will be useful,
- * but WITHOUT ANY WARRANTY; without even the implied warranty of
- * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
- * General Public License for more details.
- *
- * You should have received a copy of the GNU General Public
- * License along with this program; if not, write to the
- * Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
- * Boston, MA 02110-1301 USA.
+ /*
```
Comment (Marukesu) [inline src/Cards/UserCard.vala:1]: extra space in the begin?

File: src/Cards/UserCard.vala [hunk 6]
```diff
@@ -189,12 +177,11 @@ public class Greeter.UserCard : Greeter.BaseCard {
...
-            margin_bottom = 48,
-            orientation = Gtk.Orientation.VERTICAL
+            margin_bottom = 48
         };
-        main_grid.add (background_image);
-        main_grid.add (username_label);
-        main_grid.add (form_revealer);
+        main_grid.attach (background_image, 0, 0);
+        main_grid.attach (username_label, 0, 1);
+        main_grid.attach (form_revealer, 0, 2);
```
Comment (Marukesu) [inline src/Cards/UserCard.vala:184]: This should be a box.

Comment (lenemter) [reply src/Cards/UserCard.vala:184]: https://github.com/elementary/greeter/pull/672 requires it to be a grid, so I don't think it makes sense to change it here.

Comment (Marukesu) [reply src/Cards/UserCard.vala:184]: i don't think it requires, `Gtk.Box.pack_start()` exists, and in Gtk4, will would be using `prepend()` and `append()`.

Comment (lenemter) [reply src/Cards/UserCard.vala]: I forgot about `Gtk.Box.pack_start ()`. Thanks, I'll use it.

**Review by Marukesu — APPROVED** [review]

**Pull request merged**